lib/sasl.js

```diff
@@ -10,19 +10,34 @@ var debug = require('debug')('amqp10:sasl'),
 
     Connection = require('./connection');
 
+var saslMechanisms = {
+  PLAIN: 'PLAIN',
+  ANONYMOUS: 'ANONYMOUS',
+  NONE: 'NONE'
+};
+
 /**
- * Currently, only supports SASL-PLAIN
+ * Currently, only supports SASL ANONYMOUS or PLAIN
  *
  * @constructor
  */
-function Sasl() {
+function Sasl(mechanism) {
+  if (mechanism && !u.includes(saslMechanisms, mechanism)) {
+    throw new errors.NotImplementedError(
+      'Only SASL PLAIN and ANONYMOUS are supported.');
+  }
+  this.mechanism = mechanism;
   this.receivedHeader = false;
 }
 
+Sasl.Mechanism = saslMechanisms;
```

Comment:
I read this line as incorrect:
```
if (!u.includes(Sasl.Mechanism, saslMechanism)) {
```
in that we should be checking if the `Sasl.Mechanisms` (array/plural) "includes" the given saslMechanism (singular instance).

But I'm really not going to be so nitpicky here, if you think it's right then let's do it